hazelcast/src/main/java/com/hazelcast/config/CardinalityEstimatorConfig.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config;
+
+import static com.hazelcast.util.Preconditions.checkAsyncBackupCount;
+import static com.hazelcast.util.Preconditions.checkBackupCount;
+
+/**
+ * Configuration options for the {@link com.hazelcast.cardinality.CardinalityEstimator}
+ */
+public class CardinalityEstimatorConfig {
+
+    /**
+     * The number of sync backups per estimator
+     */
+    public static final int DEFAULT_SYNC_BACKUP_COUNT = 1;
+
+    /**
+     * The number of async backups per estimator
+     */
+    public static final int DEFAULT_ASYNC_BACKUP_COUNT = 0;
+
+    private String name = "default";
+
+    private int backupCount = DEFAULT_SYNC_BACKUP_COUNT;
+
+    private int asyncBackupCount = DEFAULT_ASYNC_BACKUP_COUNT;
+
+    private com.hazelcast.config.CardinalityEstimatorConfig.CardinalityEstimatorConfigReadOnly readOnly;
+
+    public CardinalityEstimatorConfig() {
+    }
+
+    public CardinalityEstimatorConfig(String name) {
+        this.name = name;
+    }
+
+    public CardinalityEstimatorConfig(String name, int backupCount, int asyncBackupCount) {
+        this.name = name;
+        this.backupCount = backupCount;
+        this.asyncBackupCount = asyncBackupCount;
+    }
+
+    public CardinalityEstimatorConfig(com.hazelcast.config.CardinalityEstimatorConfig config) {
+        this(config.getName(), config.getBackupCount(), config.getAsyncBackupCount());
+    }
+
+    public com.hazelcast.config.CardinalityEstimatorConfig.CardinalityEstimatorConfigReadOnly getAsReadOnly() {
+        if (readOnly == null) {
+            readOnly = new com.hazelcast.config.CardinalityEstimatorConfig.CardinalityEstimatorConfigReadOnly(this);
+        }
+        return readOnly;
+    }
+
+    /**
+     * Gets the name of the cardinality estimator.
+     *
+     * @return The name of the estimator.
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Sets the name of the cardinality estimator.
+     *
+     * @param name The name of the estimator.
+     * @return The cardinality estimator config instance.
+     */
+    public com.hazelcast.config.CardinalityEstimatorConfig setName(String name) {
+        this.name = name;
```

Comment:
I believe there should be validation on this. We don't want to have a null name.